File: 2021/09/22/game.py
"""
https://leetcode.com/problems/baseball-game/
Author: Andrew Jarombek
Date: 9/22/2021
"""

from typing import List


def calc_points(ops: List[str]) -> int:
    scores = []

    for op in ops:
        if op == "+":
            scores.append(scores[-1] + scores[-2])
        elif op == "D":
            scores.append(scores[-1] * 2)
        elif op == "C":
            scores.pop()
        else:
            scores.append(int(op))

    return sum(scores)

File: 2022/04/09/game.ts
/**
 * https://leetcode.com/problems/baseball-game/
 * @author Andrew Jarombek
 * @since 4/9/2022
 */

function game(ops: string[]): number {
    let result = [];

    for (let op of ops) {
        if (op == "C") {
            result.pop();
        } else if (op == "D") {
            result.push(result[result.length - 1] * 2);
        } else if (op == "+") {
            result.push(result[result.length - 1] + result[result.length - 2]);
        } else {
            result.push(+op);
        }
    }

    return result.reduce((acc, val) => acc + val);
}
